Initial work on new operational mode page (#2188)
hashicorp/terraform-website: content/enterprise/requirements/data-storage/vault.mdx

---
page_title: External Vault - Terraform Enterprise
---

# External Vault Requirements for Terraform Enterprise

Terraform Enterprise automatically creates an internally-managed Vault server
that stores its data in the PostgreSQL Database. We strongly recommend that
organizations use this internally-managed Vault server. However, some
organizations have specific requirements around data encryption and auditing.
Those organizations can configure Terraform Enterprise to use an external Vault

> server rather than the internally-managed Vault server.

Reviewer: ```suggestion
server rather than the internally-managed Vault server.

We only recommend using external Vault when you have experience managing Vault in production. This approach requires that you assume full responsibility for the Vault server, including sealing, unsealing, replication, etc.
```

> !> **Warning:** When using external Vault, you assume full responsibility for the Vault server. That includes sealing, unsealing, replication, etc. Do not use external Vault unless you have experience managing Vault in production.

Reviewer: ```suggestion
!> **Warning:** Do not configure multiple Terraform Enterprise instances to use the same external Vault server unless they are part of an [Active/Active](/enterprise/install/automated/active-active) installation. Doing so will result in data loss. 
```

!> **Warning:** Multiple Terraform Enterprise instances cannot use the same external Vault server unless they are a part of an Active/Active installation. Doing so will result in data loss.

~> **Important:** External Vault must be configured during initial installation and can only be changed using the [backup and restore API](/enterprise/admin/infrastructure/backup-restore).

> ### External Vault Configuration

Reviewer: ```suggestion
### External Vault Configuration

~> **Important:** You must configure External Vault during initial installation. After installation, you can only change the configuration using the [backup and restore API](/enterprise/admin/infrastructure/backup-restore).
```
